packages/vaadin-dialog/src/vaadin-dialog.js

```diff
@@ -286,22 +263,21 @@ class DialogElement extends ThemePropertyMixin(
       modeless: {
         type: Boolean,
         value: false
-      },
-
-      /** @private */
-      _oldTemplate: Object,
-
-      /** @private */
-      _oldRenderer: Object
+      }
     };
   }
 
   static get observers() {
-    return [
-      '_openedChanged(opened)',
-      '_ariaLabelChanged(ariaLabel)',
-      '_templateOrRendererChanged(_contentTemplate, renderer)'
-    ];
+    return ['_openedChanged(opened)', '_ariaLabelChanged(ariaLabel)', '_rendererChanged(renderer)'];
+  }
+
+  /** @protected */
+  connectedCallback() {
```

Comment:
I wonder if `ready()` would work better for this...especially now that there's the mutation observer, `templateRendererCallback` only needs to run once.

Reply:
Yeah, I agree, it should be better to run the template renderer callback after observers and other lifecycle callbacks have executed.

Reply:
Fixed.